File: src/main/java/com/goudourasv/data/instructors/InstructorsMapper.java
package com.goudourasv.data.instructors;

import com.goudourasv.data.institutions.InstitutionEntity;
import com.goudourasv.domain.institutions.Institution;
import com.goudourasv.domain.instructors.Instructor;

import java.util.ArrayList;
import java.util.List;


public class InstructorsMapper {
    public static Instructor toInstructor(InstructorEntity instructorEntity,List<InstitutionEntity> institutionEntities){
        List<Institution> institutions = new ArrayList<>();
        for (InstitutionEntity institutionEntity : institutionEntities) {
            Institution institution = new Institution(institutionEntity.getId(), institutionEntity.getName(), institutionEntity.getCountry(), institutionEntity.getCity());
            institutions.add(institution);
        }
        Instructor instructor = new Instructor(instructorEntity.getId(),instructorEntity.getFirstName(),instructorEntity.getLastName(),institutions);
        return instructor;
    }

    public static List<Instructor> toInstructors(List<InstructorEntity> instructorEntities){
        List<Instructor> instructors = new ArrayList<>();
        for (InstructorEntity instructorEntity : instructorEntities) {
            List<InstitutionEntity> institutionEntities = instructorEntity.getInstitutionEntities();
            Instructor instructor = toInstructor(instructorEntity,institutionEntities);
            instructors.add(instructor);
        }
        return instructors;
    }

}

File: src/main/java/com/goudourasv/data/instructors/InstructorsMapper.kt
package com.goudourasv.data.instructors

import com.goudourasv.data.institutions.InstitutionEntity
import com.goudourasv.domain.institutions.Institution
import com.goudourasv.domain.instructors.Instructor

fun InstructorEntity.toInstructor(institutionEntities: List<InstitutionEntity>): Instructor {
    //TODO check with alex
    val institutions =
        institutionEntities.map { institutionEntity ->
            Institution(

                country = institutionEntity.country,
                name = institutionEntity.name,
                id = institutionEntity.id!!,
                city = institutionEntity.city
            )
        }
    return Instructor(
        this.id!!,
        this.firstName ?: "",
        this.lastName ?: "",
        institutions
    )
}

fun List<InstructorEntity>.toInstructors(): List<Instructor> {
    val instructors = this.map { instructorEntity ->
        instructorEntity.toInstructor(instructorEntity.institutionEntities ?: emptyList())
    }
    return instructors
}